### ISSUE TITLE
Use bazel to drive the demonstrator?

### ISSUE THREAD

**Issue by proppy** [antmicro/xls-cosimulation-demonstrator#1]
Since some of the dependencies uses bazel anyway (ex: XLS), I wonder if you could simplify the demonstrator setup by leveraging bazel as its main execution/build system, ex:
- use XLS bazel workspace instead of a submodules
- add bazel rules for building the firmware
- add bazel rules for driving the simulation
- pin dependencies rather than relying on system install

Let me know what you think?

**Comment by rw1nkler** [comment]
Absolutely! I think that's a great idea and would definitely enhance the user experience.
Bazel rules for driving the simulation require more work, but the rest shouldn't be a problem.
